src/mono/wasm/Wasm.Build.Tests/BuildTestBase.cs

```diff
@@ -629,7 +633,10 @@ static void AssertRuntimePackPath(string buildOutput, string targetFramework)
             });
 
             AssertFilesExist(bundleDir, new[] { "run-v8.sh" }, expectToExist: hasV8Script);
-            AssertFilesExist(bundleDir, new[] { "icudt.dat" }, expectToExist: hasIcudt);
+            if (!string.IsNullOrEmpty(predefinedIcudt))
```

Comment:
The changes here don't look quite correct. But I think it will become clearer by adding the other tests.